services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -847,25 +847,27 @@ class HakemusService(
         excludedUserIds: Set<UUID>,
         userId: String,
     ) {
-        applicationEntity.allContactUsers().toMutableList().apply {
-            removeIf { excludedUserIds.contains(it.id) }
-            if (isNotEmpty()) {
-                val inviter =
-                    hankeKayttajaService.getKayttajaByUserId(applicationEntity.hanke.id, userId)
-                        ?: throw CurrentUserWithoutKayttajaException(userId)
-                forEach {
-                    emailSenderService.sendApplicationNotificationEmail(
-                        ApplicationNotificationData(
-                            inviter.fullName(),
-                            inviter.sahkoposti,
-                            it.sahkoposti,
-                            applicationEntity.applicationType,
-                            applicationEntity.hanke.hankeTunnus,
-                            applicationEntity.hanke.nimi,
-                        )
-                    )
-                }
-            }
+        val newContacts =
+            applicationEntity.allContactUsers().filterNot { excludedUserIds.contains(it.id) }
+        if (newContacts.isEmpty()) {
+            return
+        }
+
+        val inviter =
+            hankeKayttajaService.getKayttajaByUserId(applicationEntity.hanke.id, userId)
+                ?: throw CurrentUserWithoutKayttajaException(userId)
+
+        for (newContact in newContacts.filter { it.sahkoposti != inviter.sahkoposti }) {
```

Comment:
Why not `newContacts.filter { ... }.forEach { newContact -> ...`?  But this works just as well, so this comment can be ignored.